packages/oui-dual-list/src/dual-list.html

```diff
@@ -0,0 +1,80 @@
+        <div class="oui-dual-list__content_empty" data-ng-if="!$ctrl.sourceListLoading && !$ctrl.sourceList.length">
+            <p>{{$ctrl.sourceListEmptyLabel}}</p>
+        </div>
+        <div class="oui-dual-list__content_empty" data-ng-if="$ctrl.sourceListLoading">
+            <oui-spinner size="m"></oui-spinner>
+        </div>
+        <div class="oui-dual-list__content_body" data-ng-if="!$ctrl.sourceListLoading && $ctrl.sourceList.length">
+            <oui-search class="oui-dual-list__filter" model="$ctrl.sourceFilterKey" ng-model-options="{debounce: 500}" aria-label="search items">
+            </oui-search>
+            <div class="oui-dual-list__add-all-link" data-ng-if="$ctrl.bulkActionEnabled">
+                <oui-button variant="link" text="{{$ctrl.moveAllLabel}}" on-click="$ctrl.addAll()"></oui-button>
+            </div>
+            <ul class="oui-dual-list__content_list">
+                <li data-ng-repeat="source in $ctrl.sourceList | ouiDualListFilter:$ctrl.sourceFilterKey:$ctrl.property"
+                    data-ng-click="$ctrl.add($index, source)"
+                    class="oui-dual-list__item"
+                    title="{{$ctrl.addLabel}} {{$ctrl.getProperty(source)}}">
+                    <span data-ng-if="!$ctrl.isLoading(source)" class="oui-dual-list__item_add-icon">
+                        <i class="oui-icon oui-icon-add" aria-hidden="true"></i>
+                    </span>
+                    <span data-ng-if="!$ctrl.isLoading(source)" class="oui-dual-list__item_add-label">{{$ctrl.addLabel}}</span>
+                    <oui-spinner size="s" class="oui-dual-list__item_loading-icon" data-ng-if="$ctrl.isLoading(source)"></oui-spinner>
+                    <span class="oui-dual-list__item_title">{{$ctrl.getProperty(source)}}</span>
+                </li>
+            </ul>
+        </div>
+    </div>
+</div>
+<div class="oui-dual-list__target">
+    <div class="oui-dual-list__header">
+        <div role="heading">{{$ctrl.targetListLabel}} <span data-ng-if="!$ctrl.targetListLoading">({{$ctrl.targetList.length}})</span></div>
+        <button class="oui-dual-list__toggle-icons"
+            ng-click="$ctrl.toggleTarget()">
+            <span class="oui-icon" ng-class="{
+                    'oui-icon-chevron-up': $ctrl.isTargetOpen,
+                    'oui-icon-chevron-down': !$ctrl.isTargetOpen
+                }">
+            </span>
+        </button>
+    </div>
+    <div class="oui-dual-list__content" data-ng-if="$ctrl.isTargetOpen">
+        <div class="oui-dual-list__content_empty" data-ng-if="!$ctrl.targetListLoading && !$ctrl.targetList.length">
+            <p>{{$ctrl.targetListEmptyLabel}}</p>
+        </div>
+        <div class="oui-dual-list__content_empty" data-ng-if="$ctrl.targetListLoading">
+            <oui-spinner size="m"></oui-spinner>
+        </div>
+        <div class="oui-dual-list__content_body" data-ng-if="!$ctrl.targetListLoading && $ctrl.targetList.length">
```

Comment:
We're almost done ! There are still issue with the BEM compliance with your CSS classnames in the component.
I see that you often use modifier as an element. 

Here for example with `oui-dual-list__content_body`. `body` is not a modifier for `oui-dual-list__content`.
For example, it could be `oui-dual-list__content-body`, or `oui-dual-list__body`, or `oui-dual-list-content__body` to be compliant.

Please see https://en.bem.info/methodology/naming-convention/ for more informations
In a simple way, `block_modifier` or `block___element_modifier` but not `block___element_element` :)